SimplePayment.Test/ModelTests.cs

```diff
@@ -0,0 +1,124 @@
+
+            Assert.AreEqual(RemoveWhiteSpace(paymentResponseJson), paymentResponseSerialized);
+        }
+
+        [Test]
+        public void TestOrderDetailsModel()
+        {
+            var orderDetailsJson = ReadJson("OrderDetails");
+            var orderDetails = JsonConvert.DeserializeObject<OrderDetails>(orderDetailsJson);
+            var orderDetailsSerialized = JsonConvert.SerializeObject(orderDetails);
+
+            Assert.AreEqual(orderDetails.Merchant, "PUBLICTESTHUF");
+            Assert.AreEqual(orderDetails.OrderRef, "101010515363456734591");
+            Assert.AreEqual(orderDetails.Customer, "v2 START Tester");
+            Assert.AreEqual(orderDetails.CustomerEmail, "sdk_test@otpmobil.com");
+            Assert.AreEqual(orderDetails.Language, "HU");
+            Assert.AreEqual(orderDetails.Currency, Currency.HUF);
+            Assert.AreEqual(orderDetails.Total, 100);
+            Assert.AreEqual(orderDetails.TwoStep, true);
+            Assert.AreEqual(orderDetails.Salt, "d471d2fb24c5a395563ff60f8ba769d1");
+            Assert.AreEqual(orderDetails.Methods[0], "CARD");
+            Assert.AreEqual(orderDetails.Invoice.ToString(), JsonConvert.DeserializeObject<BillingDetails>(ReadJson("BillingDetails")).ToString());
+            Assert.AreEqual(orderDetails.Delivery.ToString(), JsonConvert.DeserializeObject<BillingDetails>(ReadJson("BillingDetails")).ToString()); // It has equal properties as BillingDetails
+            Assert.AreEqual(orderDetails.TimeOut, DateTime.Parse("2018-09-07T20:51:13+00:00"));
+            Assert.AreEqual(orderDetails.Url, "http://simplepaytestshop.hu/back.php");
+            Assert.AreEqual(orderDetails.SDKVersion, "SimplePay_PHP_SDK_2.0_180906");
+            Assert.AreEqual(orderDetails.OrderItems[0].ToString(), JsonConvert.DeserializeObject<OrderItem>(ReadJson("OrderItem")).ToString());
+        }
+
+        [Test]
+        public void TestOrderItemModel()
+        {
+            var orderItemJson = ReadJson("OrderItem");
+            var orderItem = JsonConvert.DeserializeObject<OrderItem>(orderItemJson);
+            var orderItemSerialized = JsonConvert.SerializeObject(orderItem);
+
+            Assert.AreEqual(orderItem.Ref, "Nagy Usa Körút");
+            Assert.AreEqual(orderItem.Title, "szép út");
+            Assert.AreEqual(orderItem.Description, "nagyon szép");
+            Assert.AreEqual(orderItem.Amount, 1);
+            Assert.AreEqual(orderItem.Price, 959000);
+            Assert.AreEqual(orderItem.Tax, 210000);
+        }
+        private string RemoveWhiteSpace(string json)
+        {
+            return Regex.Replace(json, @"(""[^""\\]*(?:\\.[^""\\]*)*"")|\s+", "$1");
+        }
+        private string ReadJson(string jsonFile)
+        {
+            string folderPath = Directory.GetCurrentDirectory().Replace("\\bin\\Debug\\netcoreapp3.1", "");
+            var jsonResult = File.ReadAllText($@"{folderPath}\TestJsonFiles\{jsonFile}Json.txt");
+
```

Comment:
Ennél egy picit lehetne változtatni rajta, elképzelhető nekem, hogy ebből gondjai lesznek majd, de amúgy simán lehet, hogy nem.

Ha kijelölöd az összes txt file-t a testjsonfiles folderben, akkor jobb klikk -> properties -> és ott a Copy to output directory-t be kéne állítani Copy always-ra. 
Utána a Test project .csproj-ba meg kell nézni, hogy berakta-e őket, és elmentette-e (ha nem, rámenteni), és azt is fel kell rakni gitre.
Ha ez megvan, akkor itt ebben a file beolvasásban nem is kell a folderpath, simán meg lehet adni csak úgy, hogy $@"TestJsonFiles \ {jsonFile}Json.txt"  

Mert így mindig másolja majd magával buildkor a file-okat és megmarad a relatív helye, szóval nem kell a foldert még külön nézni.